src/Functions/parseDateTime.cpp

```diff
@@ -0,0 +1,1747 @@
+        {"aug", {"ust", 8}},
+        {"sep", {"tember", 9}},
+        {"oct", {"ober", 10}},
+        {"nov", {"ember", 11}},
+        {"dec", {"ember", 12}},
+    };
+
+    /// key: month, value: total days of current month if current year is leap year.
+    constexpr Int32 leapDays[] = {0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+
+    /// key: month, value: total days of current month if current year is not leap year.
+    constexpr Int32 normalDays[] = {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+
+    /// key: month, value: cumulative days from January to current month(inclusive) if current year is leap year.
+    constexpr Int32 cumulativeLeapDays[] = {0, 31, 60, 91, 121, 152, 182, 213, 244, 274, 305, 335, 366};
+
+    /// key: month, value: cumulative days from January to current month(inclusive) if current year is not leap year.
+    constexpr Int32 cumulativeDays[] = {0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334, 365};
+
+    /// key: year, value: cumulative days from epoch(1970-01-01) to the first day of current year(exclusive).
+    constexpr Int32 cumulativeYearDays[]
+        = {0,     365,   730,   1096,  1461,  1826,  2191,  2557,  2922,  3287,  3652,  4018,  4383,  4748,  5113,  5479,  5844,  6209,
+           6574,  6940,  7305,  7670,  8035,  8401,  8766,  9131,  9496,  9862,  10227, 10592, 10957, 11323, 11688, 12053, 12418, 12784,
+           13149, 13514, 13879, 14245, 14610, 14975, 15340, 15706, 16071, 16436, 16801, 17167, 17532, 17897, 18262, 18628, 18993, 19358,
+           19723, 20089, 20454, 20819, 21184, 21550, 21915, 22280, 22645, 23011, 23376, 23741, 24106, 24472, 24837, 25202, 25567, 25933,
+           26298, 26663, 27028, 27394, 27759, 28124, 28489, 28855, 29220, 29585, 29950, 30316, 30681, 31046, 31411, 31777, 32142, 32507,
+           32872, 33238, 33603, 33968, 34333, 34699, 35064, 35429, 35794, 36160, 36525, 36890, 37255, 37621, 37986, 38351, 38716, 39082,
+           39447, 39812, 40177, 40543, 40908, 41273, 41638, 42004, 42369, 42734, 43099, 43465, 43830, 44195, 44560, 44926, 45291, 45656,
+           46021, 46387, 46752, 47117, 47482, 47847, 48212, 48577, 48942, 49308, 49673};
+
+    Int64 numLiteralChars(const char * cur, const char * end)
+    {
+        bool found = false;
+        Int64 count = 0;
+        while (cur < end)
+        {
+            if (*cur == '\'')
+            {
+                if (cur + 1 < end && *(cur + 1) == '\'')
+                {
+                    count += 2;
+                    cur += 2;
+                }
+                else
+                {
+                    found = true;
+                    break;
+                }
+            }
+            else
+            {
+                ++count;
+                ++cur;
+            }
+        }
+        return found ? count : -1;
+    }
+
+    struct Date
+    {
+        Int32 year = 1970;
+        Int32 month = 1;
+        Int32 day = 1;
+        std::vector<Int32> day_of_month_values;
```

Comment:
It is not immediately clear what `day_of_month_values` represents. It is a vector, but why would a Date have a list of days?

Reply:
`day_of_month_values` and `day_of_year_values` represents all parsed day_of_month/day_of_year values from a single string.
For example:  input = "2 30 10 2", format = "M d d M".  The reason why we need them is making sure all day_of_month/day_of_year values is valid. 

But it seems unnecessary because we always used the last appeared value as day_of_month or day_of_year. They are removed now.